currying/README.md

```diff
@@ -0,0 +1,194 @@
+---
+layout: pattern
+title: Currying
+folder: currying
+permalink: /patterns/currying/
+categories: Functional
+language: en
+tags: 
+ - Decoupling
+---
```

Comment:
Please implement the yaml frontmatter according to https://github.com/iluwatar/java-design-patterns/wiki/01.-How-to-contribute